packages/core/test/install/gitBranchLockfile.test.ts

```diff
@@ -0,0 +1,119 @@
+import fs from 'fs'
+import path from 'path'
+import { getCurrentBranch } from '@pnpm/git-utils/test/utils/mock'
+import { prepareEmpty, preparePackages } from '@pnpm/prepare'
+import { install, mutateModules } from '@pnpm/core'
+import { testDefaults } from '../utils'
+import { WANTED_LOCKFILE } from '@pnpm/constants'
+import { ProjectManifest } from '@pnpm/types'
+import writeYamlFile from 'write-yaml-file'
+
+test('install with git-branch-lockfile = true', async () => {
+  const project = prepareEmpty()
+
+  const branchName: string = 'main-branch'
+  getCurrentBranch.mockReturnValue(branchName)
+
+  const opts = await testDefaults({
+    useGitBranchLockfile: true,
+  })
+
+  await install({
+    dependencies: {
+      'is-positive': '^3.0.0',
+    },
+  }, opts)
+
+  const lockfileDir: string = project.dir()
+  expect(fs.existsSync(path.join(lockfileDir, `pnpm-lock.${branchName}.yaml`))).toBe(true)
+  expect(fs.existsSync(path.join(lockfileDir, WANTED_LOCKFILE))).toBe(false)
```

Comment:
This should work as well

```suggestion
  expect(fs.existsSync(`pnpm-lock.${branchName}.yaml`)).toBe(true)
  expect(fs.existsSync(WANTED_LOCKFILE)).toBe(false)
```